tests/models/test_admin.py

```diff
@@ -1,156 +1,180 @@
-        assert True == self.admin.is_librarian(self._default_library)
-        assert True == self.admin.is_librarian(other_library)
-
-    def test_can_see_collection(self):
-        # This collection is only visible to system admins since it has no libraries.
-        c1 = self._collection()
-
-        # This collection is visible to libraries of its library.
-        c2 = self._collection()
-        c2.libraries += [self._default_library]
-
-        # The admin has no roles yet.
-        assert False == self.admin.can_see_collection(c1);
-        assert False == self.admin.can_see_collection(c2);
-
-        self.admin.add_role(AdminRole.SYSTEM_ADMIN)
-        assert True == self.admin.can_see_collection(c1)
-        assert True == self.admin.can_see_collection(c2)
-
-        self.admin.remove_role(AdminRole.SYSTEM_ADMIN)
-        self.admin.add_role(AdminRole.SITEWIDE_LIBRARY_MANAGER)
-        assert False == self.admin.can_see_collection(c1);
-        assert True == self.admin.can_see_collection(c2);
-
-        self.admin.remove_role(AdminRole.SITEWIDE_LIBRARY_MANAGER)
-        self.admin.add_role(AdminRole.SITEWIDE_LIBRARIAN)
-        assert False == self.admin.can_see_collection(c1);
-        assert True == self.admin.can_see_collection(c2);
-
-        self.admin.remove_role(AdminRole.SITEWIDE_LIBRARIAN)
-        self.admin.add_role(AdminRole.LIBRARY_MANAGER, self._default_library)
-        assert False == self.admin.can_see_collection(c1);
-        assert True == self.admin.can_see_collection(c2);
-
-        self.admin.remove_role(AdminRole.LIBRARY_MANAGER, self._default_library)
-        self.admin.add_role(AdminRole.LIBRARIAN, self._default_library)
-        assert False == self.admin.can_see_collection(c1);
-        assert True == self.admin.can_see_collection(c2);
-
-        self.admin.remove_role(AdminRole.LIBRARIAN, self._default_library)
-        assert False == self.admin.can_see_collection(c1);
-        assert False == self.admin.can_see_collection(c2);
+
+def test_password_hashed(db_session, create_admin_user):
+    # GIVEN: An Admin user
+    # WHEN:  The Admin password is set to "password"
+    # THEN:  Check the hashed password starts with $2a$
+    admin = create_admin_user(db_session)
+    admin.password = "password"
+
+    pytest.raises(NotImplementedError, lambda: admin.password)
+    assert admin.password_hashed.startswith('$2a$')
+
+def test_with_password(db_session, create_admin_user):
+    # GIVEN: An Admin user
+    # WHEN:  There is no password set
+    # THEN:  The Admin.with_password query is empty
+    assert [] == Admin.with_password(db_session).all()
+
+    admin = create_admin_user(db_session)
+    assert [] == Admin.with_password(db_session).all()
+
+    # WHEN: A password is set
+    # THEN: The Admin.with_password query contains users
+    admin.password = "password"
+    assert [admin] == Admin.with_password(db_session).all()
+
+    admin2 = create_admin_user(db_session, email="admin2@nypl.org")
```

Comment:
Given that this is open source code run by folks other than NYPL, I wonder if it's worth creating a way to set the domain that'll be used in test email addresses. It only really matters if a test potentially triggers an actual email message (since we don't want to get spammed by outside test runners), but I could see it being useful.

Reply:
The alternative (for tests that need a domain or address whose structure is valid even if it doesn't really resolve) would probably be to go with [`example.com`](https://en.wikipedia.org/wiki/Example.com), or a fake domain using the newer [`.example`](https://en.wikipedia.org/wiki/.example) or [`.test`](https://en.wikipedia.org/wiki/.test) top level domains IANA came up with a while back.

Reply:
I agree that example.com is better (makes it clear to the reader the address is a stand-in) but since unit tests should never send actual email, changing this can wait.